CONTRIBUTING.md

```diff
@@ -108,6 +108,164 @@ To learn about the viewer, run:
+In order to update reference with the new image, run with `UPDATE_SNAPSHOTS=1` environment variable set.
+
+Use `pixi run snapshots` to compare the results of all failed tests in Rerun.
+
+##### Guidelines for writing image comparison tests
+
+* avoid! Whenever **possible** prefer regular Rust tests or `insta` snapshot tests over image comparison tests.
+* images should…
+  * …be checked in as LFS file
+  * …depict exactly what's tested and nothing else
+  * …have a low resolution to avoid growth in repo size
+  * …have a low comparison threshold to avoid the test passing despite unwanted differences
+
+##### Why does CI / another computer produce a different image?
+
+First check whether the difference is due to a change in enabled rendering features, potentially due to difference in hardware (/software renderer) capabilitites - our setup generally avoids this by enforcing the same set of features, but this may happen nonetheless.
+
+However, smaller discrepancies may be caused by a variety of implementation details depending on the concrete GPU/driver (even between different versions of the same driver).
+For instance:
+* multi-sample anti-aliasing
+  * sample placement and sample resolve steps are implementation defined
+  * alpha-to-coverage algorithm/pattern can wary wildly between implementations
+  * by default we render without anti-aliasing
+* texture filtering
+  * different implementations may apply different optimizations *even* for simple linear texture filtering
+* out of bounds texture access (via `textureLoad`)
+  * implementations are free to return indeterminate values instead of clamping
+* floating point evaluation, for details see [WGSL spec § 15.7. Floating Point Evaluation](https://www.w3.org/TR/WGSL/#floating-point-evaluation). Notably:
+  * rounding mode may be inconsistent
+  * floating point math "optimizations" may occur
+    * depending on output shading language, different arithmetic optimizations may be performed upon floating point operations even if they change the result
+  * floating point denormal flush
+    * even on modern implementations, denormal float values may be flushed to zero
+  * `NaN`/`Inf` handling
+    * whenever the result of a function should yield `NaN`/`Inf`, implementations may free to yield an indeterminate value instead
+  * builtin-function function precision & error handling (trigonometric functions and others)
+* [partial derivatives (dpdx/dpdx)](https://www.w3.org/TR/WGSL/#dpdx-builtin)
+  * implementations are free to use either `dpdxFine` or `dpdxCoarse`
+* [...]
+
+
+Whenever you can't avoid these problems there's two types of thresholds you can tweak:
+* threshold for when a pixel is considered different (see [`egui_kittest::SnapshotOptions::threshold`])
+* how many pixels are allowed to differ (see [`HarnessExt::snapshot_with_broken_pixels_threshold`])
+  TODO(emilk/egui#5683): this should be natively supported by kittest
```

Comment:
This is a wonderful section that would be great to have in the `egui_kittest` README.md. Then we could just link to it from here instead

Reply:
I'll create a PR against egui_kittest!